lib/rules/aria-conditional-attr.json

```diff
@@ -0,0 +1,13 @@
+{
+  "id": "aria-conditional-attr",
+  "matches": "aria-allowed-attr-matches",
+  "tags": ["cat.aria", "wcag2a", "wcag412", "EN-301-549", "EN-9.4.1.2"],
+  "actIds": ["5c01ea"],
+  "metadata": {
+    "description": "Ensures ARIA attributes are used as described for the element's role",
+    "help": "ARIA attributes must be used as described by the element's role"
```

Comment:
"certain scenarios" feels very vague. This checks that if the ARIA says the role has certain restrictions on when/how an attribute can be used, those restrictions are followed.

Reply:
I think I'm getting tripped up on `as described in the specification of the element's role` since what we are doing to validate aria-checked is not defined in any specification (instead the spec calls out to _not_ use `aria-checked` on an `input type="checkbox"`). But I don't want to keep bikesheding this, so going to approve and you can feel free to merge or update the message.